### PR TITLE
[Merged by Bors] - Disable XDR in codec + fixed applying scale codec in beacon and fetch

### PR DESCRIPTION
**Pull request by evt**
## Motivation
Part of #3271

## Changes
disable XDR in codec + fixed applying scale codec in beacon and fetch

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: beacon/beacon.go [hunk 1]
```diff
@@ -922,7 +937,15 @@ func buildSignedProposal(ctx context.Context, signer signing.Signer, epoch types
 	return signature
 }
 
-func signMessage(signer signing.Signer, message interface{}, logger log.Log) []byte {
+func signFirstVotingMessage(signer signing.Signer, message *FirstVotingMessageBody, logger log.Log) []byte {
```
Comment (countvonzero) [inline beacon/beacon.go]: what's the diff btwn signFirstVotingMessage and signFollowingVotingMessage?

File: beacon/beacon.go [hunk 2]
```diff
@@ -795,7 +800,12 @@ func (pd *ProtocolDriver) sendFirstRoundVote(ctx context.Context, epoch types.Ep
 		types.FirstRound,
 		log.String("message", m.String()))
 
-	pd.sendToGossip(ctx, pubsub.BeaconFirstVotesProtocol, m)
+	serialized, err := codec.Encode(&m)
```
Comment (countvonzero) [inline beacon/beacon.go:808]: do you want to move 803-806 into pd.sendToGossip()? the code here and below are repetitive

File: beacon/beacon_test.go [hunk 3]
```diff
@@ -730,41 +730,6 @@ func TestBeacon_buildProposal(t *testing.T) {
 	}
 }
 
-func TestBeacon_signMessage(t *testing.T) {
```
Comment (countvonzero) [inline beacon/beacon_test.go:733]: why rm this test?

File: beacon/beacon.go (hunk 1 above)
Comment (evt) [reply beacon/beacon.go]: they accept different message types. Scale codec doesn't work with `interface{}`. Alternatively we can move `codec.Encode` to the caller side but then only `signer.Sign` is left and no point in a function :)

File: beacon/beacon.go (hunk 2 above)
Comment (evt) [reply beacon/beacon.go:808]: nope, I don't :) scale codec doesn't work with `interface{}`

File: beacon/beacon_test.go (hunk 3 above)
Comment (evt) [reply beacon/beacon_test.go:733]: gonna add separate tests for different message types. scale codec doesn't work with `interface{}`

File: beacon/beacon.go (hunk 1 above)
Comment (countvonzero) [reply beacon/beacon.go]: right. in that case maybe just get rid of signXXXMessage all together?

File: beacon/beacon_test.go (hunk 3 above)
Comment (countvonzero) [reply beacon/beacon_test.go:733]: i am not sure this test is meaningful anyway....

Comment (evt) [reply beacon/beacon_test.go:733]: true, I'll think if I can make it :)

File: beacon/beacon.go (hunk 1 above)
Comment (evt) [reply beacon/beacon.go]: and done

**Comment by evt** [comment]
bors try

File: beacon/beacon.go (hunk 1 above)
Comment (dshulyak) [reply beacon/beacon.go]: you could replace interface{} with codec.Encodable (which should be an alias for scale.Encodable)

File: codec/codec.go [hunk 4]
```diff
@@ -6,46 +6,27 @@ import (
...
 
-func init() {
-	// xdr will fail with overflow if slice size is larger than 1mb
-	// see BenchmarkInvalidLength
-	xdr.SliceLimit = 1 << 20
-}
-
 // Encodable is an interface that must be implemented by a struct to be encoded.
-type Encodable interface{}
+type Encodable interface {
```
Comment (dshulyak) [inline codec/codec.go]: maybe Encodable = scale.Encodable

**Review by dshulyak — APPROVED** [review]

**Comment by evt** [comment]
bors try

**Comment by evt** [comment]
bors merge

**Comment by evt** [comment]
bors merge

**Pull request closed without merging**